InfiniTAM/CMakeLists.txt

```diff
@@ -85,24 +86,20 @@ message(INFO "OpenNI found: " ${OPENNI_FOUND})
 add_subdirectory(Engine)
 
 add_executable(InfiniTAM_cli InfiniTAM_cli.cpp)
-target_link_libraries(InfiniTAM_cli Engine)
-target_link_libraries(InfiniTAM_cli Utils)
+target_link_libraries(InfiniTAM_cli Engine Utils)
 IF(WITH_CUDA)
   target_link_libraries(InfiniTAM_cli ${CUDA_LIBRARIES})
 ENDIF()
 add_executable(InfiniTAM InfiniTAM.cpp)
-target_link_libraries(InfiniTAM Engine)
-target_link_libraries(InfiniTAM Utils)
+target_link_libraries(InfiniTAM Engine Utils)
 IF(WITH_CUDA)
   target_link_libraries(InfiniTAM ${CUDA_LIBRARIES})
 ENDIF()
 
-add_executable(InfiniTAM_node InfiniTAM_node.cpp)
-target_link_libraries(InfiniTAM_node Engine)
-target_link_libraries(InfiniTAM_node Utils)
+cs_add_executable(infinitam_ros_node infinitam_ros_node.cpp)
+target_link_libraries(infinitam_ros_node Engine Utils)
 IF(WITH_CUDA)
-  target_link_libraries(InfiniTAM_node ${CUDA_LIBRARIES})
+  target_link_libraries(infinitam_ros_node ${CUDA_LIBRARIES})
 ELSE()
-  target_link_libraries(InfiniTAM_node)
-ENDIF()
-
+  target_link_libraries(infinitam_ros_node)
```

Comment:
the correct way of doing it would be:

```Cmake
...
cs_add_executable(infinitam_ros_node infinitam_ros_node.cpp)
IF(WITH_CUDA)
  target_link_libraries(infinitam_ros_node Engine Utils ${CUDA_LIBRARIES})
ELSE()
  target_link_libraries(infinitam_ros_node Engine Utils)
```

or you don't do the if, else part as if cuda if cuda is not found ${CUDA_LIBRARIES} should just be the empty string:

```Cmake
...
cs_add_executable(infinitam_ros_node infinitam_ros_node.cpp)
target_link_libraries(infinitam_ros_node Engine Utils ${CUDA_LIBRARIES})
```

(But the first version is probably a bit nicer to read)